src/util/promiseRetry.ts

```diff
@@ -0,0 +1,32 @@
+type PromiseGenerator = () => Promise<any>;
+
+const DEFAULT_RETRY_ATTEMPTS = 3;
+const DEFAULT_RETRY_DELAY = 100;
+
+const wait = (delay: number): Promise<undefined> =>
+  new Promise((resolve): number => setTimeout(resolve, delay));
+
+const retry = (
+  func: PromiseGenerator,
+  times: number = DEFAULT_RETRY_ATTEMPTS,
+  delay: number = DEFAULT_RETRY_DELAY
+): Promise<any> =>
+  new Promise(
+    (resolve, reject): void => {
+      func()
+        .then(resolve)
+        .catch(
+          (err: Error): Promise<any> | void => {
+            if (--times > 0) {
+              return wait(delay)
+                .then(retry.bind(null, func, times, delay))
```

Comment:
it seems to me that
```
.then(() => retry(func, times, delay))
```
is more intuitive than using the `.bind()`, but that's just my $.02